### ISSUE TITLE
[syncfusion_flutter_pdfviewer] crash when tries to cast object as PdfDictionary?

### ISSUE THREAD

**Issue by alexgarbarev** [syncfusion/flutter-widgets#1247]
I'm getting thousands of crash reports (11K reports from 20 users for last 7 days) from Flutter Web users.

The crash itself looks like: **TypeError: Instance of 'minified:as': type 'minified:as' is not a subtype of type 'minified:aN?'**

Decoded stack trace is:
```
org-dartlang-sdk:///dart-sdk/lib/_internal/js_runtime/lib/js_helper.dart 1195:37                                                                                   wrapException
org-dartlang-sdk:///dart-sdk/lib/_internal/js_shared/lib/rti.dart 1257:3                                                                                           _failedAsCheck
org-dartlang-sdk:///dart-sdk/lib/_internal/js_shared/lib/rti.dart 1252:3                                                                                           _generalNullableAsCheckImplementation
../../../../../../.pub-cache/hosted/pub.dev/syncfusion_flutter_pdf-20.4.54/lib/src/pdf/implementation/exporting/pdf_text_extractor/font_structure.dart 1098:23     FontStructure.fontEncoding
../../../../../../.pub-cache/hosted/pub.dev/syncfusion_flutter_pdf-20.4.54/lib/src/pdf/implementation/exporting/pdf_text_extractor/font_structure.dart 2598:9      FontStructure.decodeTextExtraction
../../../../../../.pub-cache/hosted/pub.dev/syncfusion_flutter_pdf-20.4.54/lib/src/pdf/implementation/exporting/pdf_text_extractor/image_renderer.dart 630:16      ImageRenderer._renderTextElementWithLeading
../../../../../../.pub-cache/hosted/pub.dev/syncfusion_flutter_pdf-20.4.54/lib/src/pdf/implementation/exporting/pdf_text_extractor/image_renderer.dart 373:17      ImageRenderer.renderAsImage
../../../../../../.pub-cache/hosted/pub.dev/syncfusion_flutter_pdf-20.4.54/lib/src/pdf/implementation/exporting/pdf_text_extractor/pdf_text_extractor.dart 317:5   PdfTextExtractor._getTextLine
../../../../../../.pub-cache/hosted/pub.dev/syncfusion_flutter_pdf-20.4.54/lib/src/pdf/implementation/exporting/pdf_text_extractor/pdf_text_extractor.dart 207:14  PdfTextExtractor._extractTextLines
../../../../../../.pub-cache/hosted/pub.dev/syncfusion_flutter_pdf-20.4.54/lib/src/pdf/implementation/exporting/pdf_text_extractor/pdf_text_extractor.dart 126:12  CanvasRenderBox.findTextWhileHover
```

It looks like it tries to cast types incorrectly in `font_structure.dart 1098:23`. 
Can someone check a valid fix for that? Thanks!

Library version:
```syncfusion_flutter_pdfviewer: ^20.4.50``` 
> Unfortunately,  I cannot update package, because it now depends on old `device_info_plus ^8.1.0`, while other packages depends on `device_info_plus ^9.0.0`

Flutter version:
``` Flutter version 3.10.3 on channel stable```

**Comment by deepika134** [comment]
We tried to reproduce the reported issue with provided details, but the PDF document loaded properly without any crash in our side. For your reference we have attached the sample (along with the output screen record) and it can be downloaded from the attachment. If you still reproduce the reported issue on your side, kindly share the issue reproducing sample with steps/ video. This will be helpful for us to provide prompt solution at earliest.
[Sample_1247.zip](https://github.com/syncfusion/flutter-widgets/files/11718398/Sample_1247.zip)
